src/CameraZoomPlugin.cc

```diff
@@ -352,31 +393,91 @@ void CameraZoomPlugin::PreUpdate(
   if (!comp)
     return;
 
-  if (!this->impl->zoomChanged)
+  if (this->impl->zoomChanged) {
+    // Only calculate goal once each time zoom is changed.
+    const auto requestedZoomCmd = this->impl->zoomCommand.load();
+    const auto clampedZoomCmd = std::clamp(requestedZoomCmd,
+      this->impl->minZoom, this->impl->maxZoom);
+    if (std::abs(requestedZoomCmd - clampedZoomCmd) >
+      std::numeric_limits<double>::epsilon())
+    {
+      gzwarn << "Requested zoom command of " << requestedZoomCmd
+        << " has been clamped to " << clampedZoomCmd << ".\n";
+    }
+    this->impl->goalHfov =  this->impl->curHfov / clampedZoomCmd;
+    this->impl->zoomChanged = false;
+  }
+
+  // Goal is achieved, nothing to update.
+  if (std::abs(this->impl->goalHfov - this->impl->curHfov) <
+    std::numeric_limits<double>::epsilon())
     return;
 
   // Update component.
   sdf::Sensor &sensor = comp->Data();
   sdf::Camera *cameraSdf = sensor.CameraSensor();
+  assert(cameraSdf);  // Halt on debug mode intentionally.
```

Comment:
This will abort in release if the pointer is invalid. That should never happen (i.e. a plugin failure must not take out the entire application).

Reply:
Thanks, assert is removed.